cairo/ethereum/cancun/trie.cairo

```diff
@@ -238,18 +242,26 @@ func common_prefix_length(a: Bytes, b: Bytes) -> felt {
     return result;
 }
 
-func nibble_list_to_compact(x: Bytes, is_leaf: bool) -> Bytes {
+func nibble_list_to_compact{range_check_ptr: felt}(x: Bytes, is_leaf: bool) -> Bytes {
     alloc_locals;
     let (local compact) = alloc();
+    local range_check_ptr_end;
 
     if (x.value.len == 0) {
         assert [compact] = 16 * (2 * is_leaf.value);
         tempvar result = Bytes(new BytesStruct(compact, 1));
         return result;
     }
 
-    local remainder = nondet %{ ids.x.value.len % 2 %};
-    assert remainder * (1 - remainder) = 0;
+    // TODO: add a way to patch `nondet %{ ids.x.value.len % 2 %};` in test runner
```

Comment:
looking at the compiled program, you'd see a hint like

```
'memory[fp + 1] = to_felt_or_relocatable(ids.x.value.len % 2)'
```

so you don't need to rewrite the hint, just use this string in the test

Reply:
done with https://github.com/kkrt-labs/keth/issues/214